doc/design/program.md

```diff
@@ -120,6 +120,22 @@ message AttrDesc {
 }
 ```
 
+## ProgramDesc and ExecutionPlan
+
+The goal of `ProgramDesc` is to describe **what** the user wants to calculate, and the goal of `ExecutionPlan` is to specify **how** to calculate it.
+
+For example, the `ExecutionPlan` has OP placement information to indicate which device the OP will run, but the `ProgramDesc` does not have this information since currently our Python API does not support manually pinning an OP onto a type of device (e.g., GPU or FPGA). On the other hand, the `ProgramDesc` should have information about if an OP belongs to an optimizer, this information is provided by the user and helps to place the OPs onto the parameter servers, but the `ExecutionPlan` does not have this information.
```

Comment:
Maybe we also updte the describe of Op placement: https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/refactor/parameter_server.md#graph-converter, in the newer design, op placement includ device information.

Reply:
Thanks! Will do.

Reply:
Do we have two independent concept: trainer & pserver, or we only have one concept worker and the role is decided by the subgraph it receives?

Reply:
Only have one concept worker and the role is decided by the subgraph it receives.